addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1020,18 +1118,24 @@ var SnippetsMenu = Widget.extend({
 
         // Auto-selects text elements with a specific class and remove this
         // on text changes
-        this.$document.on('click.snippets_menu', '.o_default_snippet_text', function (ev) {
-            $(ev.target).closest('.o_default_snippet_text').removeClass('o_default_snippet_text');
-            $(ev.target).selectContent();
-            $(ev.target).removeClass('o_default_snippet_text');
-        });
-        this.$document.on('keyup.snippets_menu', function () {
-            var range = Wysiwyg.getRange(this);
-            $(range && range.sc).closest('.o_default_snippet_text').removeClass('o_default_snippet_text');
+        this.$document.on('click.snippets_menu', '.o_default_snippet_text', (ev) => {
+            ev.preventDefault();
+            ev.stopPropagation();
+            ev.stopImmediatePropagation();
+            const autoSelectDefaultText = async (context) => {
+                const $target = $(ev.target);
+                const $defaultSnippetText = $target.closest('.o_default_snippet_text');
+                await this.editorHelpers.removeClass(context, $defaultSnippetText[0], 'o_default_snippet_text');
+                const nodes = this.editorHelpers.getNodes($target[0]);
+                this.wysiwyg.editor.selection.select(nodes[0], nodes[nodes.length - 1]);
+            };
+            this.wysiwyg.editor.execCommand(autoSelectDefaultText);
         });
 
         const $autoFocusEls = $('.o_we_snippet_autofocus');
-        this._activateSnippet($autoFocusEls.length ? $autoFocusEls.first() : false);
+        if ($autoFocusEls.length) {
+            this._activateSnippet($autoFocusEls.first());
+        }
```

Comment:
As we were discussing the `activateSnippet(false)` conundrum, I looked around for changes regarding that. This seems wrong then, it should deactivate the current snippet instead.